### PR TITLE
Use function in framework to enable driver verifier

### PR DESCRIPTION
**Pull request by suqinhuang**
1. use function in framework to enable driver verifier
2. no need to enable driver in linux guest
3. update code base on avocado framework

Signed-off-by: Suqin Huang <shuang@redhat.com>

### CODE REVIEW

**Comment by suqinhuang** [comment]
ID: 1346204

**Comment by suqinhuang** [comment]
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.rng_bat

File: qemu/tests/cfg/rng_bat.cfg [hunk 1]
```diff
@@ -7,19 +7,11 @@
...
         read_rng_cmd  = "X:\random_%PROCESSOR_ARCHITECTURE%.exe"
-        enable_verifier_cmd = "verifier.exe /standard /driver viorng.sys"
-        driver_verifier_cmd = "verifier.exe /querysettings"
-        driver_name = "viorng.sys"
+        driver_name = "viorng"
         rng_data_rex = "0x\w"
-        need_reboot = yes
     Linux:
         session_cmd_timeout = 360
         read_rng_cmd  = "dd if=/dev/hwrng  bs=1 count=10 2>/dev/null|hexdump"
```
Comment (PyLearner) [inline qemu/tests/cfg/rng_bat.cfg]: use /dev/hwrng , the command will hang there, replace it with /dev/random ,then works fine.

File: qemu/tests/rng_bat.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+from virttest import funcatexit
+from avocado.core import exceptions
+from avocado.utils import process
 
 
-@error.context_aware
+@error_context.context_aware
 def run(test, params, env):
     """
     Qemu virtio-rng device test:
```
Comment (pingl2016) [inline qemu/tests/rng_bat.py]: Need to update the doc string, as part of steps is removed.

File: qemu/tests/rng_bat.py [hunk 3]
```diff
@@ -47,61 +49,44 @@ def set_winutils_letter(cmd, session, params):
             vol = "%s:" % vol
         return cmd.replace("X:", vol)
 
-    driver_name = params.get("driver_name")
     rng_data_rex = params.get("rng_data_rex", r".*")
     dev_file = params.get("filename_passthrough")
     timeout = float(params.get("login_timeout", 360))
     rng_dll_register_cmd = params.get("rng_dll_register_cmd")
```
Comment (pingl2016) [inline qemu/tests/rng_bat.py]: where is rng_dll_register_cmd defined?

File: qemu/tests/rng_bat.py [hunk 4]
```diff
@@ -47,61 +49,44 @@ def set_winutils_letter(cmd, session, params):
...
-                        "enable_verifier_cmd",
-                        "driver_verifier_cmd"]
-    session_cmds = (set_winutils_letter(params.get(key), session, params)
-                    for key in session_cmds_key)
-    read_rng_cmd, enable_verifier_cmd, driver_verifier_cmd = session_cmds
 
     if dev_file:
-        error.context("Check '%s' used by qemu" % dev_file, logging.info)
+        error_context.context("Check '%s' used by qemu" % dev_file, logging.info)
         if not is_dev_used_by_qemu(dev_file, vm_pid):
```
Comment (pingl2016) [inline qemu/tests/rng_bat.py]: according to the case RHEL7-11030, need to check rng pci device and hardware RNG core before reading data from  /dev/random in linux guest. shall we need to add the steps?

File: qemu/tests/rng_bat.py (hunk 3 above)
Comment (suqinhuang) [inline qemu/tests/rng_bat.py]: it may be used in old version

**Comment by CongLi** [comment]
@suqinhuang 

Please check https://github.com/avocado-framework/avocado-vt/pull/591/files and update this PR.

**Comment by suqinhuang** [comment]
@CongLi updated, could you review again.

Thanks

**Comment by CongLi** [comment]
Test failed.

10:59:47 DEBUG| Traceback (most recent call last):
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-35.0-py2.7.egg/virttest/funcatexit.py", line 29, in run_exitfuncs
10:59:47 DEBUG|     func(*targs, **kargs)
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-35.0-py2.7.egg/virttest/utils_test/qemu.py", line 174, in clear_win_driver_verifier
10:59:47 DEBUG|     ignore_all_errors=True)
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1209, in cmd
10:59:47 DEBUG|     print_func, safe)
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1136, in cmd_status_output
10:59:47 DEBUG|     o = self.cmd_output(cmd, timeout, internal_timeout, print_func, safe)
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1050, in cmd_output
10:59:47 DEBUG|     self.read_nonblocking(0, timeout)
10:59:47 DEBUG|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 677, in read_nonblocking
10:59:47 DEBUG|     end_time = time.time() + timeout
10:59:47 DEBUG| TypeError: unsupported operand type(s) for +: 'float' and 'VM'
10:59:47 ERROR| Error in clear_win_driver_verifier: unsupported operand type(s) for +: 'float' and 'VM'

**Comment by suqinhuang** [comment]
https://github.com/avocado-framework/avocado-vt/pull/600

**Comment by suqinhuang** [comment]
PASS 1-smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.rng_random.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.rng_bat

File: qemu/tests/rng_bat.py [hunk 5]
```diff
@@ -47,61 +48,43 @@ def set_winutils_letter(cmd, session, params):
...
     vm.verify_alive()
     vm_pid = vm.get_pid()
-    session = vm.wait_for_login(timeout=timeout)
-    session_cmds_key = ["read_rng_cmd",
-                        "enable_verifier_cmd",
-                        "driver_verifier_cmd"]
-    session_cmds = (set_winutils_letter(params.get(key), session, params)
-                    for key in session_cmds_key)
-    read_rng_cmd, enable_verifier_cmd, driver_verifier_cmd = session_cmds
+    driver_name = params["driver_name"]
```
Comment (pingl2016) [inline qemu/tests/rng_bat.py]: for linux guest, "driver_name" is not defined.

File: qemu/tests/rng_bat.py [hunk 6]
```diff
@@ -47,61 +47,61 @@ def set_winutils_letter(cmd, session, params):
...
+        if not re.search(r"%s" % driver_name, output, re.M):
+            msg = "Verify device driver failed, "
+            msg += "guest report driver is %s, " % output
+            msg += "expect is '%s'" % driver_name
+            raise exceptions.TestFail(msg)
 
-    error.context("Read virtio-rng device to get random number", logging.info)
+    error_context.context("Read virtio-rng device to get random number",
+                          logging.info)
+    session = vm.wait_for_login(timeout=timeout)
```
Comment (pingl2016) [inline qemu/tests/rng_bat.py]: for linux guest, session is still opened.

Comment (suqinhuang) [inline qemu/tests/rng_bat.py]: then no need to check it's Windows or linux guest.

**Comment by pingl2016** [comment]
Acked-by: Ping Li pingl@redhat.com

**Comment by PyLearner** [comment]
(1/2) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u3.rng_random.raw.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.rng_bat: PASS

test pass

LGTM ,thanks.

**Comment by ngu-niny** [comment]
Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**